test_crates/enum_no_repr_variant_discriminant_changed/new/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+// Explicit discriminant changed values. By doing so, it changed the implicit
+// discriminant's value as well, should be reported.
+pub enum ExplicitAndImplicitDiscriminantsAreChanged {
+    First = 2,
+    Second,
+    Third = 5,
+}
+
+// Discriminant changed values. Having #[non_exhaustive] on the enum should not have any effect
+// on the *API* breakage, should still be reported.
+#[non_exhaustive]
+pub enum DiscriminantIsChanged {
+    First = 1,
+}
+
+// Discriminant changed values and the variant is also marked as `non_exhaustive`.
+// This now implies that the enum is no longer `well-defined`, which means that a numeric
+// cast is no longer possible on the enum, should not be reported.
```

Comment:
```suggestion
// This now implies that the discriminant is no longer `well-defined`, which means that a numeric
// cast is no longer possible on the enum. Should not be reported.
```
Please try to use precise, clear language and good grammar as much as possible. Long sentences with lots of commas are hard to read.